crypto/fipsmodule/hmac/hmac.c

```diff
@@ -64,34 +64,91 @@
 
 #include "../../internal.h"
 
+typedef int (*HashInit)(void *);
+typedef int (*HashUpdate)(void *, const void*, size_t);
+typedef int (*HashFinal)(uint8_t *, void*);
```

Comment:
Why this function doesn't take the ctx as the first parameter?

Reply:
Because that's how [libcrypto defines it](https://www.openssl.org/docs/man1.1.1/man3/SHA1_Final.html) :-(